src/components/common/table/TableDensitySettingsButton.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useMemo, useState } from 'react';
+
+import { Box, Popover, Typography, useTheme } from '@mui/material';
+import { Dropdown, DropdownItem, Tooltip } from '@terraware/web-components';
+import { TableDensityType } from '@terraware/web-components/components/table/types';
+
+import Button from 'src/components/common/button/Button';
+import { useLocalization, useUser } from 'src/providers';
+import strings from 'src/strings';
+
+export type DensitySettingsProp = {
+  density?: TableDensityType;
+  onChange?: (density: TableDensityType) => void;
+};
+
+const DensitySettings = ({ density, onChange }: DensitySettingsProp): JSX.Element => {
+  const { activeLocale } = useLocalization();
+  const theme = useTheme();
+  const { updateUserPreferences, userPreferences } = useUser();
+  const tableDensity: TableDensityType = useMemo(
+    () => density ?? (userPreferences['tableDensity'] as TableDensityType) ?? 'comfortable',
+    [density, userPreferences]
+  );
+
+  const saveTableDensity = (newDensity: TableDensityType) => {
+    if (onChange) {
+      onChange(newDensity);
+    }
+    updateUserPreferences({ tableDensity: newDensity });
+  };
+
+  const options: DropdownItem[] = useMemo(
+    () => [
+      {
+        label: strings.COMPACT,
+        value: 'compact',
+      },
+      {
+        label: strings.COMFORTABLE,
+        value: 'comfortable',
+      },
+      {
+        label: strings.ROOMY,
+        value: 'roomy',
+      },
+    ],
+    [activeLocale]
```

Comment:
This should be throwing an eslint error because you aren't using the dep within the memo fn. I wonder if the recent react changes broke this. Either way, you should do something like `activeLocale ? [...] : []` so that when these linting errors are working again this won't cause issue

Reply:
Can you do it like this? It _should_ be impossible for `activeLocale` to be undefined, but in the event it is, better to show nothing as opposed to options with no label. Sorry the `activeLocale` stuff is so cumbersome, Karthik and I were talking about baking it into strings or making a hook that handles it
```
  const options = useMemo(
    () => activeLocale ? [
      {
        label: strings.DENSITY_COMPACT, 
        value: 'compact',
      },
      {
        label: strings.DENSITY_COMFORTABLE,
        value: 'comfortable',
      },
      {
        label: strings.DENSITY_ROOMY,
        value: 'roomy',
      },
    ] : [],
    [activeLocale]
);
```